java/client/src/main/java/glide/api/models/GlideString.java

```diff
@@ -0,0 +1,90 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.Getter;
+
+// TODO docs for the god of docs
+public class GlideString {
+
+    @Getter private byte[] bytes;
+    private String string = null;
+
+    /** Flag whether possibility to convert to string was checked. */
+    private final AtomicBoolean conversionChecked = new AtomicBoolean(false);
+
+    private GlideString() {}
+
+    public static GlideString of(String string) {
+        var res = new GlideString();
+        res.string = string;
+        res.bytes = string.getBytes(StandardCharsets.UTF_8);
```

Comment:
Can we avoid calling `getBytes` here and call it on demand (and storing the result in `res.bytes`)?